src/state/slices/stakingDataSlice/selectors.ts

```diff
@@ -169,14 +218,157 @@ export const selectRewardsByAccountSpecifier = createDeepEqualOutputSelector(
+    if (!stakingData || !stakingData.rewards) return {}
+
+    const rewards = stakingData.rewards.reduce(
+      (acc: Record<string, chainAdapters.cosmos.Reward[]>, current: ValidatorReward) => {
+        if (!acc[current.validator.address]) {
+          acc[current.validator.address] = []
+        }
+
+        acc[current.validator.address].push(
+          ...current.rewards.filter(x => ASSET_ID_TO_DENOM[x.assetId] === denom)
+        )
+
+        return acc
+      },
+      {}
+    )
+
+    return rewards
+  }
+)
+
 export const selectRewardsAmountByDenom = createSelector(
   selectRewardsByAccountSpecifier,
+  selectValidatorAddress,
   selectDenom,
-  (rewardsByAccountSpecifier, denom): string => {
+  (rewardsByAccountSpecifier, validatorAddress, denom): string => {
     if (!rewardsByAccountSpecifier.length) return ''
 
     const rewards = rewardsByAccountSpecifier.find(rewards => rewards.denom === denom)
 
     return rewards?.amount || ''
   }
 )
+
+export const selectAllValidators = createDeepEqualOutputSelector(
+  selectStakingData,
+  stakingData => stakingData.byValidator
+)
+
+export const selectSingleValidator = createSelector(
+  selectStakingData,
+  selectValidatorAddress,
+  (stakingData, validatorAddress) => {
+    return stakingData.byValidator[validatorAddress] || null
+  }
+)
+
+export const selectNonloadedValidators = createSelector(
+  selectStakingDataByAccountSpecifier,
+  selectAllValidators,
+  (stakingData, allValidators): string[] => {
+    if (!stakingData) return []
+    const initialValidatorsAddresses = stakingData.delegations
+      ? stakingData.delegations.map(x => x.validator.address)
+      : []
+    if (stakingData.undelegations) {
+      initialValidatorsAddresses.push(...stakingData.undelegations.map(x => x.validator.address))
+    }
+
+    if (stakingData.rewards) {
+      initialValidatorsAddresses.push(...stakingData.rewards.map(x => x.validator.address))
+    }
+
+    const uniqueValidatorAddresses = [...new Set(initialValidatorsAddresses)]
+    return uniqueValidatorAddresses.filter(x => allValidators[x] === undefined)
+  }
+)
+
+export const selectUndelegationsAmountByValidatorAddress = memoize(
+  (
+    allUndelegationsEntries: Record<string, chainAdapters.cosmos.UndelegationEntry[]>,
+    validatorAddress: string
+  ) => {
+    return get(
+      allUndelegationsEntries,
+      validatorAddress,
+      [] as chainAdapters.cosmos.UndelegationEntry[]
+    )
+      .reduce((acc: BigNumber, undelegationEntry: chainAdapters.cosmos.UndelegationEntry) => {
+        acc = acc.plus(bnOrZero(undelegationEntry.amount))
+        return acc
+      }, bnOrZero(0))
+      .toString()
+  },
+  (
+    allUndelegationsEntries: Record<string, chainAdapters.cosmos.UndelegationEntry[]>,
+    validatorAddress: string
+  ) =>
+    get(allUndelegationsEntries, validatorAddress, [] as chainAdapters.cosmos.UndelegationEntry[])
+)
+
+export const selectRewardsAmountByValidatorAddress = memoize(
```

Comment:
Nitpick and naming pedantry but that's not really a state selector anymore as it's not a createSelector `(state: ReduxState) => state.slice.access.possibly.derived.data` function (which is what a selector really is if you were to recompute all the way down to the initial selector), but rather a function that takes some arbitrary arguments - whether or not the arguments are slices isn't relevant.

```suggestion
export const getRewardsAmountByValidatorAddress = memoize(
```

Reply:
Fixed